summoner-cli/src/Summoner/Template/Nix.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE QuasiQuotes    #-}
+
+module Summoner.Template.Nix
+       ( nixFiles
+       ) where
+
+import NeatInterpolation (text)
+
+import Summoner.Default (defaultGHC)
+import Summoner.GhcVer (GhcVer (..))
+import Summoner.Settings (NixPkgSet (..), Settings (..), defaultNixPkgSet)
+import Summoner.Tree (TreeFs (..))
+
+import qualified Data.Text as T
+
+nixCompiler :: GhcVer -> Text
```

Comment:
I guess it can be moved to `GhcVer` module. (probably we can make it without explicit pattern matching, but this is not urgent)